<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="N. Pollet" />


<title>Metagenome</title>

<script src="site_libs/header-attrs-2.5/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Metatetard</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="LifeStages.html">LifeStages</a>
</li>
<li>
  <a href="Adult.html">Adult</a>
</li>
<li>
  <a href="Diet.html">Diet</a>
</li>
<li>
  <a href="EarlyLate.html">EarlyLate</a>
</li>
<li>
  <a href="Transmission.html">Transmission</a>
</li>
<li>
  <a href="Metagenome.html">Metagenome</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Metagenome</h1>
<h4 class="author">N. Pollet</h4>
<h4 class="date">13/12/2020</h4>

</div>

<div id="TOC">
<ul>
<li><a href="#analysis-of-xenopus-metagenome-and-metatranscriptome">Analysis of Xenopus metagenome and metatranscriptome</a></li>
<li><a href="#section"></a></li>
<li><a href="#read-krona.tab-output-files-from-matam">Read krona.tab output files from matam</a>
<ul>
<li><a href="#visualisation-of-metagenome-bins-using-vizbin">Visualisation of metagenome bins using vizbin</a></li>
</ul></li>
</ul>
</div>

<div id="analysis-of-xenopus-metagenome-and-metatranscriptome" class="section level3">
<h3>Analysis of Xenopus metagenome and metatranscriptome</h3>
<div id="taxonomic-affiliations-using-full-length-rrna-gene-sequences" class="section level4">
<h4>Taxonomic affiliations using full-length rRNA gene sequences</h4>
<p>-Krona plot of <a href="Figure_S9_metaDNA_MATAM_krona.html">taxonomic affiliations obtained from MATAM on metaDNA</a></p>
<pre><code># Run matam on metaDNA
DBDIR=/home/Databases/MATAM
matam_assembly.py -d $DBDIR/SILVA_128_SSURef_NR95 -i reads4matam/metaDNA_reads.fastq --cpu 24 --max_memory 50000 -v --perform_taxonomic_assignment -o matam_output</code></pre>
<p>You can find the <a href="metaD%20NA_MATAM_final_assembly.fa">assembled sequences here</a></p>
<p>-Krona plot of <a href="Figure_S10_metaDNA.phyloFlash.html">phyloFlash results on metaDNA</a></p>
<pre><code># Run phyloflash on metaDNA
phyloFlash.pl -lib metaDNA -read1 READ_QC/metatetard_DNA/final_pure_reads_1.fastq -read2 READ_QC/metatetard_DNA/final_pure_reads_2.fastq -CPUs 24 -everything</code></pre>
<p>You can find the <a href="metaDNA.phyloFlash.all.final.fasta">assembled sequences here</a></p>
<p>-Krona plot of <a href="Figure_S11_metaRNA_MATAM_krona.html">taxonomic affiliations obtained from MATAM on metaRNA</a></p>
<pre><code># Run matam on metaRNA
# This is a very intensive computation so you may need to tune the max_memory and coverage_threshold according to your needs.
DBDIR=/home/Databases/MATAM
matam_assembly.py -d $DBDIR/SILVA_128_SSURef_NR95 -i reads4matam/metaRNA_reads.fastq --cpu 72 --max_memory 500000 -v --perform_taxonomic_assignment --coverage_threshold 500 --min_read_node 2 --min_overlap_edge 2 -o matam_output</code></pre>
<p>You can find the <a href="metaRNA_MATAM_final_assembly.fa">assembled sequences here</a></p>
<p>-Krona plot of <a href="Figure_S12_metaRNA.phyloFlash.html">phyloFlash results on metaRNA</a></p>
<pre><code># Run phyloflash on metaRNA
phyloFlash.pl -lib metaRNA -read1 final_pure_reads_1.fastq -read2 final_pure_reads_2.fastq -CPUs 24 -everything -trusted Xt_metatrin.fasta</code></pre>
<p>You can find the <a href="metaRNA.phyloFlash.all.final.fasta">assembled sequences here</a></p>
</div>
</div>
<div id="section" class="section level3">
<h3></h3>
</div>
<div id="read-krona.tab-output-files-from-matam" class="section level1">
<h1>Read krona.tab output files from matam</h1>
<pre><code>#
library(ggplot2)
library(dplyr)
library(ggbump)
library(tidyr)

source(&#39;theme_npgray.R&#39;)

#Importing the krona.tab file produced by matam
matam_mt&lt;-read.table(&quot;metatranscriptomic/mt.krona.tab&quot;,header=FALSE,sep=&quot;\t&quot;)
matam_mg&lt;-read.table(&quot;metagenomic/mg.krona.tab&quot;,header=FALSE,sep=&quot;\t&quot;)

#Assign headers - MT for the numbers of reads
matam_mt&lt;-setNames(matam_mt,c(&quot;MT&quot;,&quot;Kingdom&quot;,&quot;Phylum&quot;,&quot;Class&quot;,&quot;Order&quot;,&quot;Family&quot;,&quot;Genus&quot;))
matam_mg&lt;-setNames(matam_mg,c(&quot;MG&quot;,&quot;Kingdom&quot;,&quot;Phylum&quot;,&quot;Class&quot;,&quot;Order&quot;,&quot;Family&quot;,&quot;Genus&quot;))
#Group by Phylum
mt_by_phylum &lt;- matam_mt %&gt;% group_by(Phylum) %&gt;% summarise(MT=sum(MT))
mg_by_phylum &lt;- matam_mg %&gt;% group_by(Phylum) %&gt;% summarise(MG=sum(MG))

#Join the (two) datasets
matam&lt;-full_join(mg_by_phylum,mt_by_phylum,by=&quot;Phylum&quot;)
#Replace NA by 0
matam&lt;-matam %&gt;% mutate_if(is.numeric, ~replace(., is.na(.), 0))

#Compute percentages
matam&lt;-matam %&gt;% mutate(MetaRNA=100*(MT/sum(MT)),MetaDNA=100*(MG/sum(MG)))

#For working on read numbers
#  matam %&gt;% pivot_longer(c(&#39;MT&#39;,&#39;MG&#39;),names_to=&quot;Experiment&quot;,values_to=&quot;Nb_reads&quot;)

#Working only on percentages
matam&lt;-matam %&gt;% select(Phylum,MetaRNA,MetaDNA) %&gt;% pivot_longer(c(&#39;MetaRNA&#39;,&#39;MetaDNA&#39;),names_to=&quot;Experiment&quot;,values_to=&quot;Percentage&quot;)

# Plotting a stacked bar plot
colors&lt;-c(&quot;Actinobacteria&quot;=&quot;#810027&quot;,&quot;Bacteroidetes&quot;=&quot;#92c5de&quot;,&quot;Cyanobacteria/Chloroplast&quot;=&quot;#f7f7f7&quot;,&quot;Deferribacteres&quot;=&quot;#f4a582&quot;,&quot;Firmicutes&quot;=&quot;#053061&quot;,&quot;Fusobacteria&quot;=&quot;#80cdc1&quot;,&quot;Lentisphaerae&quot;=&quot;#ffffff&quot;,&quot;Proteobacteria&quot;=&quot;#d6604d&quot;,&quot;Synergistetes&quot;=&quot;#2166ac&quot;,&quot;Verrucomicrobia&quot;=&quot;#fddbc7&quot;,&quot;unclassified&quot;=&quot;#f0f0f0&quot;)

ggplot() +
geom_bar(data = matam,
aes(x = Experiment, y =Percentage, fill = Phylum),
colour = &#39;black&#39;, width = 0.6, stat=&quot;identity&quot;) +
geom_sigmoid(data = tidyr::spread(matam, Experiment, Percentage),
colour = &quot;black&quot;,
direction=&quot;x&quot;,
smooth=6,
aes(x = 1 + 0.6/2,
xend = 2 - 0.6/2,
y = 100-cumsum(MetaDNA),
yend = 100-cumsum(MetaRNA)))+
scale_fill_manual(values=colors)+
theme_npgray()+coord_flip()

ggsave(&quot;meta_matam_abundance.pdf&quot;,width=16,height=18,units=&quot;cm&quot;)</code></pre>
<div id="visualisation-of-metagenome-bins-using-vizbin" class="section level3">
<h3>Visualisation of metagenome bins using vizbin</h3>
<p>-Vizbin of <a href="Script_vizbin_DAS.html">DASTool binning results</a></p>
<div id="metabolic-pathways-inference" class="section level4">
<h4>Metabolic pathways inference</h4>
<p>-Krona plot of <a href="Figure_S13_metaDNA.kegg.minpath.html">KEGG metabolic pathways</a></p>
<p>-Krona plot of <a href="Figure_S14_metaDNA.metacyc.minpath.html">Metacyc metabolic pathways</a></p>
<p>-iPath3 visualization is available <a href="https://pathways.embl.de/ipath3.cgi?s=pWbci4bo871W8Qm8XKF&amp;map=metabolic">here</a></p>
</div>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
